dod-advana/gamechanger-data
# Section Parsing: Navy, CJCS, EO, and Title Documents

This PR builds off of the PR for [section-parse-dod](https://github.com/dod-advana/gamechanger-data/pull/248).

## Major Changes:
- Added the following section parser classes (all are children of `ParserDefinition`):
    - `NavyParser`
        - extracts **Purpose**, **Responsibilities**, and **Policy** sections
        - supported doc types: OPNAVINST, OPNAVNOTE, BUMEDINST, MILPERSMAN, RESPERSMAN, BUPERSINST, SECNAVINST, SECNAVNOTE, MCO, NAVMC, COMNAVRESFORCOMINST, COMNAVRESFORCOMNOTE, BUMEDNOTE, ALNAV, MCRP, MCBUL, FMFRP
    - `CJCSParser`
        - extracts **Purpose** and **Responsibilities** sections
        - supported doc types: CJCSI, CJCSM, CJCSN, CJCS GDE
    - `EOParser`
        - extracts **Purpose** sections
        - supported doc types: EO
    - `USCodeParser`
        - extracts **Purpose** sections
        - supported doc types: Title

## How to Test
- See test instructions in the "Tests" section [here](https://github.com/dod-advana/gamechanger-data/blob/section-parse-navy-cjcs-eo-title/common/document_parser/lib/section_parse/README.md).
- See unit test files [here](https://github.com/dod-advana/gamechanger-data/tree/section-parse-navy-cjcs-eo-title/common/document_parser/lib/section_parse/tests/unit).
- See integrated test files [here](https://github.com/dod-advana/gamechanger-data/tree/section-parse-navy-cjcs-eo-title/common/document_parser/lib/section_parse/tests/integrated).
